python/gui/forwarding/consumer.py

```diff
@@ -233,14 +294,13 @@ async def listen(self):
         """
         Listen for messages from the target connection and send them through the caller connection
         """
-        if self.__connection is None:
-            await self._connect_to_target()
-
-        async for message in self.__connection:
-            if isinstance(message, bytes):
-                await self.send(bytes_data=message)
-            else:
-                await self.send(message)
+        while True:
```

Comment:
Since this in unbounded, we should add the return type hint of `-> typing.NoReturn`

Reply:
Not a bad idea. Might also be handy to detail how this is being managed and why to make it clear that this is a dependent operation that will be interrupted and won't necessarily create an infinite blocking loop.